client/asset/interface.go

```diff
@@ -671,6 +676,8 @@ type Contract struct {
 type Redemption struct {
 	// Spends is the AuditInfo for the swap output being spent.
 	Spends *AuditInfo
+	// SwapDetails is the swap details.
+	SwapDetails *dex.SwapContractDetails
```

Comment:
I'd like to refactor this later. `AuditInfo` and  `SwapContractDetails` are identical except for an extra `Contract` field. But I didn't want to mess with the btc/dcr interfaces too much in the PR.